src/prettytable/prettytable.py

```diff
@@ -2423,9 +2423,9 @@ class TableHandler(HTMLParser):
     def __init__(self, **kwargs) -> None:
         HTMLParser.__init__(self)
         self.kwargs = kwargs
-        self.tables = []
-        self.last_row = []
-        self.rows = []
+        self.tables: list[list] = []
+        self.last_row: list[str] = []
+        self.rows: list[Any] = []
```

Comment:
It's good to avoid `Any` if possible, as it's like having no typing.

So `self.rows` is a list of tuples:

```suggestion
        self.rows: list[tuple] = []
```

But it's also good practice to give more accurate types for things like tuples, lists and dicts.

The next step after this PR would be to add the `--strict` flag to mypy, which will complain about these, but then there are over 300 errors! Anyway, we can maybe do a few of these here. (And there's an automated tool that can help with many of those, but that can also wait.)

So this tuple is a list (of table data) and then a Boolean:

```suggestion
        self.rows: list[tuple[list, bool]] = []
```

Ideally we should also say what can be in the list. This can be things like strings, ints, floats, I guess Booleans too, but maybe others? Maybe we can worry about enumerating that later (and when we do we can use a [_type alias_](https://docs.python.org/3/library/typing.html#type-aliases), like a variable for types).

So I think `list[tuple[list, bool]]` is enough for here :)

Reply:
This is all super useful, I can't thank you enough for your personal feedback!

Reply:
@hugovk Okay, I'll read about the `--strict` flag, thank you.

Reply:
> @hugovk This is all super useful, I can't thank you enough for your personal feedback!

Happy to help, I'm also learning stuff!

> @hugovk Okay, I'll read about the `--strict` flag, thank you.

It might also be that we don't want to enable `--strict` on the CI and fix all of those warnings, maybe just a few. But one step at a time, and we can do some with https://github.com/JelleZijlstra/autotyping :)